cfn-resources/search-index/cmd/resource/resource.go

```diff
@@ -152,43 +158,41 @@ func newSearchIndex(currentModel *Model) (*admin.ClusterSearchIndex, error) {
 }
 
 // convertToAnySlice function converts a slice of map[string]interface{} to a slice of interface{}
-func convertToAnySlice(input []map[string]interface{}) []interface{} {
-	result := make([]interface{}, len(input))
-	for i, v := range input {
-		result[i] = v
+func convertToAnySlice(input []string) ([]interface{}, error) {
+	var result []interface{}
+
+	for _, jsonStr := range input {
+		var data interface{}
+		err := json.Unmarshal([]byte(jsonStr), &data)
+		if err != nil {
+			return nil, err
+		}
+		result = append(result, data)
 	}
-	return result
+
+	return result, nil
 }
 
-func newTokenizerModel(tokenizer map[string]interface{}) admin.ApiAtlasFTSAnalyzersTokenizer {
-	result := admin.ApiAtlasFTSAnalyzersTokenizer{}
-	if v, ok := tokenizer["maxGram"]; ok {
-		result.MaxGram = admin.PtrInt(cast.ToInt(v))
-	}
-	if v, ok := tokenizer["minGram"]; ok {
-		result.MinGram = admin.PtrInt(cast.ToInt(v))
-	}
-	if v, ok := tokenizer["type"]; ok {
-		result.Type = admin.PtrString(cast.ToString(v))
-	}
-	if v, ok := tokenizer["group"]; ok {
-		result.Group = admin.PtrInt(cast.ToInt(v))
+func newTokenizerModel(tokenizer *ApiAtlasFTSAnalyzersTokenizer) admin.ApiAtlasFTSAnalyzersTokenizer {
```

Comment:
please unit test this

Reply:
i dont think we need unit testing for this one, is just a straight conversion